File: grimdb/sdk/java/src/main/java/com/grimlocker/sdk/model/SshKeyEntry.java
package com.grimlocker.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.grimlocker.sdk.Entry;
import java.util.Map;

@JsonIgnoreProperties(ignoreUnknown = true)
public class SshKeyEntry {

    @JsonProperty("id")         public String id;
    @JsonProperty("title")      public String title;
    @JsonProperty("public_key") public String publicKey;
    @JsonProperty("comment")    public String comment;
    @JsonProperty("algorithm")  public String algorithm;

    public Map<String, String> toFields() {
        return Map.of("public_key", publicKey == null ? "" : publicKey,
                      "comment",    comment == null ? "" : comment,
                      "algorithm",  algorithm == null ? "" : algorithm);
    }

    public static SshKeyEntry fromEntry(Entry e) {
        SshKeyEntry k = new SshKeyEntry();
        if (e == null) return k;
        k.id        = e.id;
        k.title     = e.title;
        k.publicKey = e.field("public_key");
        k.comment   = e.field("comment");
        k.algorithm = e.field("algorithm");
        return k;
    }

    @Override
    public String toString() {
        return "SshKeyEntry{id='" + id + "', title='" + title + "', algorithm='" + algorithm + "'}";
    }
}

File: grimdb/sdk/php/src/Entries/SshKeyEntry.php
<?php declare(strict_types=1);
namespace Grimlocker\Entries;

class SshKeyEntry
{
    public function __construct(
        public readonly string $id         = '',
        public readonly string $title      = '',
        public readonly string $publicKey  = '',
        public readonly string $privateKey = '',
        public readonly string $username   = '',
        public readonly string $passphrase = '',
    ) {}

    public static function fromEntry(Entry $e): self
    {
        return new self(
            $e->id,
            $e->title,
            $e->field('public_key'),
            $e->field('private_key'),
            $e->field('username'),
            $e->field('passphrase'),
        );
    }

    /** @return array<string,string> */
    public function toFields(): array
    {
        return [
            'public_key'  => $this->publicKey,
            'private_key' => $this->privateKey,
            'username'    => $this->username,
            'passphrase'  => $this->passphrase,
        ];
    }
}
